pkg/channelserver/channelserver.go

```diff
@@ -38,20 +38,22 @@ func GetURLAndInterval() (string, string) {
 
 }
 
-func run(driver, port, prefix string) chan error {
+func run() chan error {
```

Comment:
Instead of dropping the inputs and hard coding everything this function can accept something like `cmdArgs []string` then expand when creating the command. This would give a lot more flexibility if this does need to be called elsewhere and also allows for testing around this function. 

`config-key` `path-prefix` and `listen-address` can come in this way and the others can be left alone as they are based off settings that can be manipulated externally.